packages/nocms-plugin-image-resources/test/createImageResourceProvider.spec.js

```diff
@@ -0,0 +1,94 @@
+import assert from 'assert';
+// import { createImageResourceProvider } from '../lib/createImageResourceProvider.js';
+import { activate } from '../lib/index.js';
+import fs from 'fs';
+
+const filePaths = [
+	'test/test-images/50mm-249.jpg',
+	'test/test-images/8-tallet.jpg'
+];
+
+let resourceProviders = [];
+let pluginActivationContext;
+const fakeFiles = [];
+const writtenFiles = [];
+
+function fileToBuffer(path){
+	return new Promise((resolve, reject) => {
+		fs.readFile(path, (err, data) => {
+			if( err ) {
+				return reject(err);
+			}
+			return resolve(data);
+		});
+	});
+}
+
+function readFileFake(fakePath) {
+	return new Promise((resolve, reject) => {
+		fakeFiles.forEach( fakeFile => {
```

Comment:
You could also do something like:

```
const fakeFile = fakeFiles.find(fakeFile => fakeFile.path === fakePath);

if (!fakeFile) {
   reject('wat);
}

resolve(fakeFile.data);
```

Personally, in general I prefer to use `.find` if I am only looking for one specific index of an array, rather than `.forEach` with a scoped if.

- It also matches better with "guards" this way, rather than having the `reject` as the last solution.